### ISSUE TITLE
Unable to determine endianness for architecture 'armv6l

### ISSUE THREAD

**Issue by Dukat-Gul** [jhofstee/meta-bin-deb#3]
I have attempted to follow the necessary steps, however I perhaps am missing something crucial to get it to work.

```

pi@raspberrypi:~/venus/build$ bb> export DISTRO=raspbian-buster
pi@raspberrypi:~/venus/build$ bb> bitbake update-meta
NOTE: Bitbake server didn't start within 5 seconds, waiting for 90
NOTE: Your conf/bblayers.conf has been automatically updated.
NOTE: Started PRServer with DBfile: /home/pi/venus/persistent/raspbian-buster/prserv.sqlite3, IP: 127.0.0.1, PORT: 34913, PID: 12854
NOTE: Terminating PRServer...
NOTE: Started PRServer with DBfile: /home/pi/venus/persistent/raspbian-buster/prserv.sqlite3, IP: 127.0.0.1, PORT: 41923, PID: 12856
WARNING: Host distribution "raspbian-10" has not been validated with this version of the build system; you may possibly experience unexpected failures. It is recommended that you use a tested distribution.
ERROR: /home/pi/venus/sources/meta-victronenergy/meta-venus/recipes-deb/reprepro/reprepro.bb: Unable to determine endianness for architecture 'armv6l'
ERROR: /home/pi/venus/sources/meta-victronenergy/meta-venus/recipes-deb/reprepro/reprepro.bb: Please add your architecture to siteinfo.bbclass
ERROR: Failed to parse recipe: /home/pi/venus/sources/meta-victronenergy/meta-venus/recipes-deb/reprepro/reprepro.bb

Summary: There was 1 WARNING message shown.
Summary: There were 3 ERROR messages shown, returning a non-zero exit code.

```

Summary of the steps undertaken - as per the high level guide in the [README. ](https://github.com/jhofstee/meta-bin-deb/blob/master/README) 

Initially - Following the Steps on the Victron github page [README](https://github.com/victronenergy/venus/blob/master/README.md) ONLY up to this point;

```
Getting started
Building Venus requires a Linux. At Victron we use Ubuntu for this.

# clone this repository
git clone https://github.com/victronenergy/venus.git
cd venus

# install host packages (Debian based)
sudo make prereq
```

Then from this pages' guide I follow these steps;

```

make CONFIG=raspbian fetch-all  

Note: tried fetch-all and was prompted with the auth which obviously I dont have for the private git repo, so retried without -all)
make CONFIG=raspbian fetch

# this will update the generated meta layer
make bb
export DISTRO=raspbian-jessie
Note: also tried with 
export DISTRO=raspbian-buster
bitbake update-meta

```
However where its not clear is how and where to place the files from this (or if appropriate [VenusBot](https://github.com/venus-bot/meta-bin-deb-generated)).
So I tried the following....

```
pi@raspberrypi:~ $ git clone https://github.com/jhofstee/meta-bin-deb
Cloning into 'meta-bin-deb'...
remote: Enumerating objects: 1048, done.
remote: Counting objects: 100% (128/128), done.
remote: Compressing objects: 100% (66/66), done.
remote: Total 1048 (delta 53), reused 115 (delta 49), pack-reused 920
Receiving objects: 100% (1048/1048), 451.26 KiB | 595.00 KiB/s, done.
Resolving deltas: 100% (464/464), done.

cp -R ~/meta-bin-deb/recipes-deb ~/venus/sources/meta-victronenergy/meta-venus/
cp -R ~/meta-bin-deb/recipes-ve ~/venus/sources/meta-victronenergy/meta-venus/
cp -R ~/meta-bin-deb/recipes-test ~/venus/sources/meta-victronenergy/meta-venus/
cp -R ~/meta-bin-deb/recipes-qt ~/venus/sources/meta-victronenergy/meta-venus/
cp -R ~/meta-bin-deb/classes ~/venus/sources/meta-victronenergy/meta-venus/
cp -R ~/meta-bin-deb/conf ~/venus/sources/meta-victronenergy/meta-venus/

also tried the below;
cp -R ~/meta-bin-deb/recipes-deb ~/venus/sources/openembedded-core/meta/
cp -R ~/meta-bin-deb/recipes-ve ~/venus/sources/openembedded-core/meta/
cp -R ~/meta-bin-deb/recipes-test ~/venus/sources/openembedded-core/meta/
cp -R ~/meta-bin-deb/recipes-qt ~/venus/sources/openembedded-core/meta/
cp -R ~/meta-bin-deb/classes ~/venus/sources/openembedded-core/meta/
cp -R ~/meta-bin-deb/conf ~/venus/sources/openembedded-core/meta/


```

Any thoughts / suggestions as to where I have gone wrong... or what can be tweaked to fix this error?
thx

**Comment by jhofstee** [comment]
Hi, first of all, you typically don't want to do this yourself. The bot typically maintains this. Its output is missing in the latest release of venus, but they are available in candidate, see http://updates.victronenergy.com/feeds/venus/candidate/packages/. That does include buster, if that is what you are looking for....

**Comment by Dukat-Gul** [comment]
Thanks for responding. The reason I am attempting this path is because I tried the [victron raspbian method](https://github.com/victronenergy/venus/wiki/install-venus-packages-on-Debian) (with updates for buster)..


```
root@raspberrypi:~# echo "deb [trusted=yes] https://updates.victronenergy.com/feeds/venus/candidate/packages/raspbian-buster buster main" > /etc/apt/sources.list.d/victronenergy.list
root@raspberrypi:~# apt-get update                                                                                                                  Hit:1 http://archive.raspberrypi.org/debian buster InRelease
Ign:2 https://updates.victronenergy.com/feeds/venus/candidate/packages/raspbian-buster buster InRelease
Hit:3 http://raspbian.raspberrypi.org/raspbian buster InRelease
Hit:4 https://updates.victronenergy.com/feeds/venus/candidate/packages/raspbian-buster buster Release
Ign:5 https://updates.victronenergy.com/feeds/venus/candidate/packages/raspbian-buster buster Release.gpg
Reading package lists... Done
root@raspberrypi:~# mkdir -p /data/{conf,db}
root@raspberrypi:~# apt-get install vrmlogger
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 vrmlogger : Depends: packagegroup-ve-dbus-data-producers but it is not installable
E: Unable to correct problems, you have held broken packages.

```

however - that method does seem to work for "raspbian-jessie jessie main" as compared to buster..... as per the above paste with the error that packagegroup-ve-dbus-data-producers is not installable

**Comment by jhofstee** [comment]
I had a look why the package index is not created correctly, https://github.com/victronenergy/openembedded-core/commit/482f234e88e6ea46bfe874f997b914bce22deae9 should fix that. It should be fixed in Venus v2.70~13.

**Comment by Dukat-Gul** [comment]
Thanks. Ill survive on Jessie for now :) but am eagerly anticipating the next candidate release
